base/setup/lib/utils/partlist.c

```diff
@@ -3668,21 +3689,33 @@ WritePartitions(
     DumpPartitionTable(DiskEntry);
 #endif
 
-    /* Update the partition numbers */
+    /* Update the partition numbers and device names */
 
     /* Update the primary partition table */
     for (ListEntry = DiskEntry->PrimaryPartListHead.Flink;
          ListEntry != &DiskEntry->PrimaryPartListHead;
          ListEntry = ListEntry->Flink)
     {
         PartEntry = CONTAINING_RECORD(ListEntry, PARTENTRY, ListEntry);
+        if (!PartEntry->IsPartitioned)
+            continue;
+        ASSERT(PartEntry->PartitionType != PARTITION_ENTRY_UNUSED);
 
-        if (PartEntry->IsPartitioned)
+        /*
+         * Initialize its device partition number and its device name
+         * only if the partition was new. Note that the partition number
+         * should not change if this partition has not been deleted
+         * during repartitioning.
+         * FIXME: hbelusca: Our PartMgr currently seems to return modified
+         * numbers in the layout, this needs to be investigated and fixed.
```

Comment:
Should a Jira issued be created for this?

Reply:
Perhaps, but I'm also investigating it in some other branch so it's just more like a fixme/remember thing to myself.